third_party/NordicSemiconductor/CMakeLists.txt

```diff
@@ -0,0 +1,369 @@
+    nrfx/drivers/src/nrfx_nvmc.c
+    nrfx/drivers/src/nrfx_power.c
+    nrfx/drivers/src/nrfx_spis.c
+    nrfx/drivers/src/nrfx_systick.c
+    nrfx/hal/nrf_nvmc.c
+    nrfx/soc/nrfx_atomic.c
+)
+
+set(USB_SOURCES
+    nrfx/drivers/src/nrfx_usbd.c
+    libraries/atfifo/nrf_atfifo.c
+    libraries/atomic/nrf_atomic.c
+    libraries/usb/app_usbd.c
+    libraries/usb/app_usbd_core.c
+    libraries/usb/app_usbd_string_desc.c
+    libraries/usb/app_usbd_serial_num.c
+    libraries/usb/app_usbd_nrf_dfu_trigger.c
+    libraries/usb/nrf_dfu_trigger_usb.c
+    libraries/usb/class/cdc/acm/app_usbd_cdc_acm.c
+)
+
+set(RADIO_DRIVER_SOURCES
+    drivers/radio/fal/nrf_802154_fal.c
+    drivers/radio/mac_features/ack_generator/nrf_802154_ack_data.c
+    drivers/radio/mac_features/ack_generator/nrf_802154_ack_generator.c
+    drivers/radio/mac_features/ack_generator/nrf_802154_enh_ack_generator.c
+    drivers/radio/mac_features/ack_generator/nrf_802154_imm_ack_generator.c
+    drivers/radio/mac_features/nrf_802154_csma_ca.c
+    drivers/radio/mac_features/nrf_802154_delayed_trx.c
+    drivers/radio/mac_features/nrf_802154_filter.c
+    drivers/radio/mac_features/nrf_802154_frame_parser.c
+    drivers/radio/mac_features/nrf_802154_precise_ack_timeout.c
+    drivers/radio/fem/three_pin_gpio/nrf_fem_three_pin_gpio.c
+    drivers/radio/nrf_802154_core_hooks.c
+    drivers/radio/nrf_802154_core.c
+    drivers/radio/nrf_802154_critical_section.c
+    drivers/radio/nrf_802154_debug.c
+    drivers/radio/nrf_802154_pib.c
+    drivers/radio/nrf_802154_rssi.c
+    drivers/radio/nrf_802154_rx_buffer.c
+    drivers/radio/nrf_802154_timer_coord.c
+    drivers/radio/nrf_802154.c
+    drivers/radio/platform/coex/nrf_802154_wifi_coex_none.c
+    drivers/radio/platform/clock/nrf_802154_clock_ot.c
+    drivers/radio/platform/hp_timer/nrf_802154_hp_timer.c
+    drivers/radio/rsch/nrf_802154_rsch_crit_sect.c
+    drivers/radio/rsch/nrf_802154_rsch.c
+    drivers/radio/timer_scheduler/nrf_802154_timer_sched.c
+)
+
+set(SINGLE_PHY_SOURCES
+    drivers/radio/nrf_802154_notification_direct.c
+    drivers/radio/nrf_802154_priority_drop_direct.c
+    drivers/radio/nrf_802154_request_direct.c
+    drivers/radio/rsch/raal/single_phy/single_phy.c
+)
+
+set(RADIO_DRIVER_SOFTDEVICE_SOURCES
+    drivers/radio/nrf_802154_notification_swi.c
+    drivers/radio/nrf_802154_priority_drop_swi.c
+    drivers/radio/nrf_802154_request_swi.c
+    drivers/radio/nrf_802154_swi.c
+    drivers/radio/rsch/raal/softdevice/nrf_raal_softdevice.c
+)
+
+set(RADIO_DRIVER_SINGLE_PHY_SOURCES
+    drivers/radio/nrf_802154_notification_direct.c
+    drivers/radio/nrf_802154_priority_drop_direct.c
+    drivers/radio/nrf_802154_request_direct.c
+    drivers/radio/rsch/raal/single_phy/single_phy.c
+)
+
+if(OT_PLATFORM STREQUAL "nrf52811")
+    add_library(nordicsemi-nrf52811-sdk)
+    target_compile_definitions(nordicsemi-nrf52811-sdk
+        PRIVATE
+            ${COMMON_FLAG}
+            -DNRF52811_XXAA
```

Comment:
Merge common definitions to a variable.

Reply:
common definitions already in COMMON_FLAG